eng/versioning/external_dependencies.txt

```diff
@@ -82,6 +82,7 @@ org.apache.commons:commons-compress;1.24.0
 org.apache.avro:avro;1.11.3
 org.apache.avro:avro-maven-plugin;1.11.3
 org.apache.commons:commons-lang3;3.12.0
+org.apache.groovy:groovy;4.0.15
```

Comment:
We needed to add this because [the latest version](https://mvnrepository.com/artifact/org.spockframework/spock-core/2.4-M1-groovy-4.0) of Spock Core uses Groovy 4.0.6 :(